pyo3-bytes/src/bytes.rs

```diff
@@ -62,6 +63,66 @@ impl PyBytes {
     pub fn as_slice(&self) -> &[u8] {
         self.as_ref()
     }
+
+    /// Slice the underlying buffer using a Python slice object
+    ///
+    /// This should behave the same as Python's byte slicing:
+    ///     - `ValueError` if step is zero
+    ///     - Negative indices a-ok
+    ///     - If start/stop are out of bounds, they are clipped to the bounds of the buffer
+    ///     - If start > stop, the slice is empty
+    ///
+    /// This is NOT exposed to Python under the `#[pymethods]` impl
+    fn slice(&self, slice: &Bound<'_, PySlice>) -> PyResult<PyBytes> {
+        let len_isize = self.0.len() as isize;
+        let psi = slice.indices(len_isize)?;
```

Comment:
What does `psi` stand for? Could we maybe have a more descriptive name?

Reply:
Or, it looks like `psi` is never used apart from the next line, so it might be clearer to create a new scope
```rs
let (start, stop, step) = { 
    let indices = slice.indices(len_isize)?;
    (psi.start, psi.stop, psi.step)
};
```